packages/block-editor/src/components/link-control/index.js

```diff
@@ -313,27 +419,49 @@ function LinkControl( {
 			ref={ wrapperNode }
 			className="block-editor-link-control"
 		>
-			{ isEditingLink || ! value ? (
+			{ isResolvingLink && (
+				<div className="block-editor-link-control__loading">
+					<Spinner /> { __( 'Creating Page' ) }
+				</div>
+			) }
+
+			{ ( isEditingLink || ! value ) && ! isResolvingLink && (
 				<LinkControlSearchInput
 					value={ inputValue }
 					onChange={ onInputChange }
-					onSelect={ ( suggestion ) => {
-						onChange( { ...value, ...suggestion } );
+					onSelect={ async ( suggestion ) => {
+						if (
+							suggestion.type &&
+							CREATE_TYPE === suggestion.type
+						) {
+							// Await the promise to ensure `stopEditing` is not called prematurely.
+							await handleOnCreate( inputValue );
+						} else {
+							handleSelectSuggestion( suggestion, value )();
```

Comment:
Why is `handleSelectSuggestion` assigned as a function inside a function, if we're only ever just calling it like this?

```suggestion
							handleSelectSuggestion( suggestion, value );
```

Reply:
See https://github.com/WordPress/gutenberg/pull/19775#discussion_r375012551